### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by poberezhniuk**
- [CLICK TO SEE DEMO](https://poberezhniuk.github.io/user-managment-app/singup)

> Create user list page.

### CODE REVIEW

File: src/app/in-memory-data.service.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      },
+      {
+        id: 1,
+        name: "Isaac",
+        surname: "Newton",
+        email: "isaacnewton@gmail.com",
+        password: "gravitation"
+      },
+      {
+        id: 1,
```
Comment (alexandr2134) [inline src/app/in-memory-data.service.ts]: keep unique IDs, even for mocked data

File: src/app/singup/singup.component.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    else if (this.userControls[type].hasError("minlength"))
+      return "To short...";
+    else if (this.userControls[type].hasError("email")) return "Invalid email";
+    else if (this.userControls[type].hasError("pattern"))
+      return "Only letters and numbers";
+    else return;
+  }
+
+  addUser() {
+    if (this.userForm.invalid) {
```
Comment (alexandr2134) [inline src/app/singup/singup.component.ts]: this is smelly code. revert conditions and get rid of "return;"

File: src/app/user-fields/user-fields.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    else if (this.userControls[type].hasError("minlength"))
+      return "To short...";
+    else if (this.userControls[type].hasError("email")) return "Invalid email";
+    else if (this.userControls[type].hasError("pattern"))
+      return "Only letters and numbers";
+    else return;
+  }
+
+  addUser() {
+    if (this.userForm.invalid) {
```
Comment (alexandr2134) [inline src/app/user-fields/user-fields.component.ts]: same as: src/app/singup/singup.component.ts line:56

File: src/app/in-memory-data.service.ts (hunk 1 above)
Comment (poberezhniuk) [reply src/app/in-memory-data.service.ts]: done

File: src/app/singup/singup.component.ts (hunk 2 above)
Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: done

File: src/app/singup/singup.component.ts [hunk 4]
```diff
@@ -53,10 +53,6 @@ export class SingupComponent implements OnInit {
   }
 
   addUser() {
-    if (this.userForm.invalid) {
-      return;
-    } else {
-      this.userService.addUser(this.user);
-    }
+    this.userService.addUser(this.user);
```
Comment (alexandr2134) [inline src/app/singup/singup.component.ts]: why did you remove that validation ?

Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: because, button is enable when all fields are correct
if any field isn't invalid(length, required, email etc) button comes disabled

Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: In the following comments, I will leave more informative comments about the changes

Comment (alexandr2134) [reply src/app/singup/singup.component.ts]: you can open devtools in browser and remove "disabled" attribute manually. In such case you won't have any validation enabled and frontend will send a bad request to backend.

Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: done

File: src/app/singup/singup.component.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    else if (this.userControls[type].hasError("minlength"))
+      return "To short...";
+    else if (this.userControls[type].hasError("email")) return "Invalid email";
+    else if (this.userControls[type].hasError("pattern"))
+      return "Only letters and numbers";
+    else return;
+  }
+
+  addUser() {
+    if (!this.userForm.invalid) this.userService.addUser(this.user);
```
Comment (alexandr2134) [inline src/app/singup/singup.component.ts]: You have to go from TRUE state. And you are using  NOT FALSE check second time in a row. There is a field .valid inside of form-object.

File: src/app/singup/singup.component.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    return;
```
Comment (alexandr2134) [inline src/app/singup/singup.component.ts]: why do you need return here ?

File: src/app/top-navigation/top-navigation.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (alexandr2134) [inline src/app/top-navigation/top-navigation.component.ts]: you are using " allover the project and ' in this file. Fix that. You have to use single code-style.

File: src/app/user-fields/user-fields.component.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    const reader = new FileReader();
+    const img = event.target.files[0];
+
+    this.user.profileImg = img;
+    reader.readAsDataURL(img);
+    reader.onload = () => (this.profileImgURL = reader.result);
+  }
+
+  addUser() {
+    if (!this.userForm.invalid) this.userService.addUser(this.user);
```
Comment (alexandr2134) [inline src/app/user-fields/user-fields.component.ts]: same... revert condition. check if TRUE instead of NOT FALSE

File: src/app/singup/singup.component.ts (hunk 5 above)
Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: done

File: src/app/singup/singup.component.ts (hunk 6 above)
Comment (poberezhniuk) [reply src/app/singup/singup.component.ts]: removed

File: src/app/top-navigation/top-navigation.component.ts (hunk 7 above)
Comment (poberezhniuk) [reply src/app/top-navigation/top-navigation.component.ts]: default style for new component, i never open it before
done

File: src/app/user-fields/user-fields.component.ts (hunk 8 above)
Comment (poberezhniuk) [reply src/app/user-fields/user-fields.component.ts]: done

**Review by alexandr2134 — APPROVED** [review]
ok
